### DISCUSSION TITLE
How to register webhook

### DISCUSSION THREAD

**Discussion by snake302** [def-studio/telegraph#397] [Q&A] [answered]
Hello,

I am new to Laravel and I'm trying to understand how to 
1) register my webhook
2) validate my functions and commands 

Here is my code located in the **app/Http/Webhooks/TelegramWebhookController.php**

```
<?php

namespace App\Http\Webhooks;

use DefStudio\Telegraph\Handlers\WebhookHandler;
use DefStudio\Telegraph\Models\TelegraphChat;
use DefStudio\Telegraph\Models\TelegraphBot;
use Illuminate\Http\Request;
use Illuminate\Http\Response;

class TelegramWebhookController extends
    \DefStudio\Telegraph\Handlers\WebhookHandler
{
    protected function handleChatMessage(Stringable $text): void
    {
        $this->chat->html("Received: $text")->send();
    }
}
```

In the config I tried to keep default code, 
```
    "webhook_handler" =>
        app/Http/Webhooks/TelegramWebhookController::class,
    "debug_mode" => true,
    "report_unknown_webhook_commands" => true,

    "security" => [
        "allow_callback_queries_from_unknown_chats" => true,
        "allow_messages_from_unknown_chats" => true,
        "store_unknown_chats_in_db" => true,
    ],
```

But it's not responding to any messages, my webhook is created and registered in the Telegram Botfather, domain with https, everything is running, but Postman retuns 404 on POST, my logs at storage/laravel.log it's also empty. I unable to find any code example and it's very to hard to start...

**Comment by snake302** [comment]
I see that chat **telegraph_chats** is filling with accounts I am testing bot from that means that I have settled up everything, but not webhook registration

**Comment by fabio-ivona** [comment]
Hi, just start by launching `php artisan route:list` and check if the telegraph webook route is listed

**Reply by snake302** [reply]
Hey, thanks, yes, I see it listed as POST telegraph/webhook

**Reply by fabio-ivona** [reply]
the chat you are sending from, is it registered as telegraph chat?

Otherwise, just try to register it or temporarily enable the allow_messages_from_unknown_chats option

**Reply by snake302** [reply]
yes, allow_messages_from_unknown_chats is true, I see this chat in the DB
<img width="815" alt="Screenshot 2023-08-04 at 10 45 23" src="https://github.com/defstudio/telegraph/assets/1431423/bea2e68f-404a-4d18-adc3-447dc8489b8d">

**Comment by snake302** [comment, marked as answer]
With @fabio-ivona help I figured the issue:

1) On my server **vendor** folder were missing, so I ran composer install / composer update commands
2) I went to vendor/defstudio/telegraph/src/Controllers/WebhookController.php path and added logging 
**use Illuminate\Support\Facades\Log;
Log::debug("webhook recived $request");**
```
use Illuminate\Support\Facades\Log;

class WebhookController
{
  public function handle(Request $request, string $token): Response
  {
    Log::debug("webhook recived $request");
```
3) I figured that my laravel.log was missing 777 write rules and my TelegramWebhookController.php also were was `use Illuminate\Support\Stringable;`

Now everything is working! Thank you so much for assisting me!
